spec/osut_tests_spec.rb

```diff
@@ -3142,6 +3143,175 @@ module M
     # mini2.osm C  FullInteriorAndExteriorWithReflections  253 GJ 3.5 UMH cool
   end
 
+  it "checks for outdoor-facing roofs" do
+    translator = OpenStudio::OSVersion::VersionTranslator.new
+    expect(mod1.reset(DBG)).to eq(DBG)
+    expect(mod1.level).to eq(DBG)
+    expect(mod1.clean!).to eq(DBG)
+
+    # --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- #
+    # CASE 1: Each space has 1x outdoor-facing roof.
+    file  = File.join(__dir__, "files/osms/in/5ZoneNoHVAC.osm")
+    path  = OpenStudio::Path.new(file)
+    model = translator.loadModel(path)
+    expect(model).to_not be_empty
+    model = model.get
+
+    spaces = {}
+    roofs  = {}
+
+    model.getSpaces.each do |space|
+      space.surfaces.each do |s|
+        next unless s.surfaceType.downcase == "roofceiling"
+        next unless s.outsideBoundaryCondition.downcase == "outdoors"
+
+        expect(spaces).to_not have_key(space.nameString)
+        spaces[space.nameString] = s.nameString
+      end
+    end
+
+    expect(spaces.size).to eq(5)
+    # puts spaces
+    # "Story 1 East Perimeter Space"  : "Surface 18"
+    # "Story 1 North Perimeter Space" : "Surface 12"
+    # "Story 1 Core Space"            : "Surface 30"
+    # "Story 1 South Perimeter Space" : "Surface 24"
+    # "Story 1 West Perimeter Space"  : "Surface 6"
+
+    model.getSpaces.each do |space|
+      rufs = mod1.getRoofs(space)
+      expect(rufs.size).to eq(1)
+      ruf = rufs.first
+      expect(ruf).to be_a(OpenStudio::Model::Surface)
+      roofs[space.nameString] = ruf.nameString
+    end
+
+    expect(roofs.size).to eq(spaces.size)
+    expect(mod1.status).to be_zero
+    # puts roofs
+    # "Story 1 East Perimeter Space"  : "Surface 18"
+    # "Story 1 North Perimeter Space" : "Surface 12"
+    # "Story 1 Core Space"            : "Surface 30"
+    # "Story 1 South Perimeter Space" : "Surface 24"
+    # "Story 1 West Perimeter Space"  : "Surface 6"
+
+    spaces.each do |id, surface|
+      expect(roofs.keys).to include(id)
+      expect(roofs[id]).to eq(surface)
+    end
+
+    # --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- #
+    # CASE 2: None of the occupied spaces have outdoor-facing roofs, yet the
+    # plenum above has 4x outdoor-facing roofs (each matches 1x space ceiling).
```

Comment:
getRoofs fetches both outdoor-facing RoofCeiling surfaces, as well as those of a plenum/attic directly above.